tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/internal/DockerUtils.java

```diff
@@ -46,124 +41,208 @@
+    private static final String CONFIG_PARAMETER_TEMPLATE = "\\{\\{CONFIGURATION_PARAMETER_BLOCK}}";
+    private static final String JARS_BLOCK_TEMPLATE = "\\{\\{JARS_BLOCK}}";
+    private static final String BUNDLES_BLOCK_TEMPLATE = "\\{\\{BUNDLES_BLOCK}}";
+    private static final String ENV_BLOCK_TEMPLATE = "\\{\\{ENV_BLOCK}}";
+
+    private static final String CONFIG_BLOCK_VALUE =
+            "COPY --chown=siddhi_user:siddhi_io \\$\\{CONFIG_FILE}/ \\$\\{USER_HOME}";
+    private static final String CONFIG_PARAMETER_VALUE =
+            ", \"-Dconfig=\\$CONFIG_FILE_PATH\"";
+    private static final String JARS_BLOCK_VALUE =
+            "COPY --chown=siddhi_user:siddhi_io \\$\\{HOST_JARS_DIR}/ \\$\\{JARS}";
+    private static final String BUNDLES_BLOCK_VALUE =
+            "COPY --chown=siddhi_user:siddhi_io \\$\\{HOST_BUNDLES_DIR}/ \\$\\{BUNDLES}";
+
     private static final String RESOURCES_DIR = "resources/docker-export";
-    private static final String SIDDHI_FILES_DIR = "siddhi-files";
-    private static final String README_FILE = "README.md";
-    private static final String DOCKER_COMPOSE_FILE = "docker-compose.yml";
-    private static final String DOCKER_COMPOSE_RUNNER_FILE = "docker-compose.runner.yml";
-    private static final String PRODUCT_VERSION_TOKEN = "\\{\\{PRODUCT_VERSION}}";
-    private final ConfigProvider configProvider;
-    private DockerConfigs dockerConfigs;
+    private static final String ZIP_FILE_NAME = "siddhi-docker.zip";
+    private static final String ZIP_FILE_ROOT = "siddhi-docker/";
+    private static final String DOCKER_FILE_NAME = "Dockerfile";
+
+    private static final String JARS_DIR = "jars/";
+    private static final String BUNDLE_DIR = "bundles/";
+    private static final String APPS_DIR = "siddhi-files/";
+    private static final String CONFIG_FILE = "configurations.yaml";
 
-    public DockerUtils(ConfigProvider configProvider) {
+    private ExportAppsRequest exportAppsRequest;
 
-        this.configProvider = configProvider;
+    DockerUtils(ExportAppsRequest exportAppsRequest) {
+        this.exportAppsRequest = exportAppsRequest;
     }
 
     /**
      * Create a zip archive.
      *
-     * @param siddhiFiles List of Siddhi siddhiFiles
      * @return Zip archive file
      * @throws DockerGenerationException if docker generation fails
      */
-    public File createArchive(List<String> siddhiFiles) throws DockerGenerationException {
-        // Create <CARBON_HOME>/tmp/docker-export directory.
-        String tmpDirPath = Paths.get(Constants.CARBON_HOME, "tmp", "docker-export").toString();
-        File tmpDir = new File(tmpDirPath);
-        if (!tmpDir.exists() && !tmpDir.mkdirs()) {
-            throw new DockerGenerationException("Cannot create temporary directory at " + tmpDirPath);
-        }
+    public File createZipFile() throws DockerGenerationException {
+
+        boolean jarsChanged = false;
```

Comment:
shall we update the variable reference names as jarsAdded ? since we are not changing a previous state